components/ActionSheet/src/MDCActionSheetController.m

```diff
@@ -0,0 +1,357 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCActionSheetController.h"
+
+#import "MDCActionSheetItemTableViewCell.h"
+#import "MDCActionSheetDataSource.h"
+#import "MDCActionSheetHeaderView.h"
+#import "MaterialBottomSheet.h"
+#import "MaterialApplication.h"
+#import "MaterialTypography.h"
+
+static NSString *const kReuseIdentifer = @"BaseCell";
+
+@interface MDCActionSheetAction ()
+
+@property(nonatomic, nullable, copy) MDCActionSheetHandler completionHandler;
+
+@end
+
+@implementation MDCActionSheetAction
+
++ (instancetype)actionWithTitle:(NSString *)title
+                          image:(UIImage *)image
+                          handler:(void (^__nullable)(MDCActionSheetAction *action))handler {
+    return [[MDCActionSheetAction alloc] initWithTitle:title
+                                                 image:image
+                                               handler:handler];
+}
+
+- (instancetype)initWithTitle:(NSString *)title
+                        image:(UIImage *)image
+                      handler:(void (^__nullable)(MDCActionSheetAction *action))handler {
+  self = [super init];
+  if (self) {
+    _title = [title copy];
+    _image = [image copy];
+    _completionHandler = [handler copy];
+  }
+  return self;
+}
+
+- (id)copyWithZone:(__unused NSZone *)zone {
+  MDCActionSheetAction *action = [[self class] actionWithTitle:self.title
+                                                         image:self.image
+                                                       handler:self.completionHandler];
+  return action;
+}
+
+@end
+
+@interface MDCActionSheetController () <MDCBottomSheetPresentationControllerDelegate, UITableViewDelegate>
+
+@end
+
+@implementation MDCActionSheetController {
+  MDCActionSheetHeaderView *_header;
+  UITableViewController *_tableViewController;
+  MDCActionSheetDataSource *_dataSource;
+  NSString *_actionSheetTitle;
+  MDCBottomSheetTransitionController *_transitionController;
+  BOOL mdc_adjustFontForContentSizeCategory;
+  /**
+    Used to determine if we need to do a layout within viewWillLayoutSubviews,
+    Because we are setting the preferredContentSize we can no longer get the frame to set the sheet
+    and headers sizes after a transition. 
+   */
+  BOOL needsPreferredContentSizeUpdate;
```

Comment:
Add an underscore (I forgot that in my comment).

Reply:
Done